File: frontend/admin/packages/modules/system/src/api/system/admin/v1/base_table_archive.ts
import service from "@liujitcn/kratos-admin-core/request";
import type { Empty } from "@liujitcn/kratos-admin-system/rpc/google/protobuf/empty";
import type { BaseTableArchiveService } from "@liujitcn/kratos-admin-system/rpc/system/admin/v1/base_table_archive";
import type {
  BaseTableArchiveForm,
  CreateBaseTableArchiveRequest,
  DeleteBaseTableArchiveRequest,
  GetBaseTableArchiveRequest,
  PageBaseTableArchiveRequest,
  PageBaseTableArchiveResponse,
  SetBaseTableArchiveStatusRequest,
  UpdateBaseTableArchiveRequest
} from "@liujitcn/kratos-admin-system/rpc/system/admin/v1/base_table_archive";

const BASE_TABLE_ARCHIVE_URL = "/v1/admin/base/table-archive";

/** 表归档配置服务实现。 */
export class BaseTableArchiveServiceImpl implements BaseTableArchiveService {
  /** 分页查询归档配置。 */
  PageBaseTableArchive(request: PageBaseTableArchiveRequest): Promise<PageBaseTableArchiveResponse> {
    return service<PageBaseTableArchiveRequest, PageBaseTableArchiveResponse>({
      url: BASE_TABLE_ARCHIVE_URL,
      method: "get",
      params: request
    });
  }

  /** 查询归档配置。 */
  GetBaseTableArchive(request: GetBaseTableArchiveRequest): Promise<BaseTableArchiveForm> {
    return service<GetBaseTableArchiveRequest, BaseTableArchiveForm>({
      url: `${BASE_TABLE_ARCHIVE_URL}/${request.id}`,
      method: "get"
    });
  }

  /** 创建归档配置。 */
  CreateBaseTableArchive(request: CreateBaseTableArchiveRequest): Promise<Empty> {
    return service<BaseTableArchiveForm | undefined, Empty>({
      url: BASE_TABLE_ARCHIVE_URL,
      method: "post",
      data: request.base_table_archive
    });
  }

  /** 更新归档配置。 */
  UpdateBaseTableArchive(request: UpdateBaseTableArchiveRequest): Promise<Empty> {
    return service<BaseTableArchiveForm | undefined, Empty>({
      url: `${BASE_TABLE_ARCHIVE_URL}/${request.base_table_archive?.id ?? ""}`,
      method: "put",
      data: request.base_table_archive
    });
  }

  /** 删除归档配置。 */
  DeleteBaseTableArchive(request: DeleteBaseTableArchiveRequest): Promise<Empty> {
    return service<DeleteBaseTableArchiveRequest, Empty>({ url: `${BASE_TABLE_ARCHIVE_URL}/${request.id}`, method: "delete" });
  }

  /** 设置归档配置状态。 */
  SetBaseTableArchiveStatus(request: SetBaseTableArchiveStatusRequest): Promise<Empty> {
    return service<SetBaseTableArchiveStatusRequest, Empty>({
      url: `${BASE_TABLE_ARCHIVE_URL}/${request.id}/status`,
      method: "put",
      data: request
    });
  }
}

/** 表归档配置服务实例。 */
export const defBaseTableArchiveService = new BaseTableArchiveServiceImpl();

File: backend/internal/service/system/admin/v1/base_table_archive.go
package admin

import (
	"context"
	"fmt"

	adminv1 "github.com/liujitcn/kratos-admin/backend/api/gen/go/system/admin/v1"
	biz "github.com/liujitcn/kratos-admin/backend/internal/biz/system/admin"
	"github.com/liujitcn/kratos-core/errorsx"
	"google.golang.org/protobuf/types/known/emptypb"

	"github.com/go-kratos/kratos/v3/log"
)

// BaseTableArchiveService 管理表归档配置。
type BaseTableArchiveService struct {
	adminv1.UnimplementedBaseTableArchiveServiceServer
	baseTableArchiveCase *biz.BaseTableArchiveCase
}

// NewBaseTableArchiveService 创建表归档配置服务。
func NewBaseTableArchiveService(baseTableArchiveCase *biz.BaseTableArchiveCase) *BaseTableArchiveService {
	return &BaseTableArchiveService{baseTableArchiveCase: baseTableArchiveCase}
}

// PageBaseTableArchive 分页查询表归档配置。
func (s *BaseTableArchiveService) PageBaseTableArchive(ctx context.Context, req *adminv1.PageBaseTableArchiveRequest) (*adminv1.PageBaseTableArchiveResponse, error) {
	result, err := s.baseTableArchiveCase.PageBaseTableArchive(ctx, req)
	if err != nil {
		log.Error(fmt.Sprintf("PageBaseTableArchive %v", err))
		return nil, errorsx.WrapInternal(err, "查询归档配置失败")
	}
	return result, nil
}

// GetBaseTableArchive 查询表归档配置。
func (s *BaseTableArchiveService) GetBaseTableArchive(ctx context.Context, req *adminv1.GetBaseTableArchiveRequest) (*adminv1.BaseTableArchiveForm, error) {
	result, err := s.baseTableArchiveCase.GetBaseTableArchive(ctx, req.GetId())
	if err != nil {
		log.Error(fmt.Sprintf("GetBaseTableArchive %v", err))
		return nil, errorsx.WrapInternal(err, "查询归档配置失败")
	}
	return result, nil
}

// CreateBaseTableArchive 创建表归档配置。
func (s *BaseTableArchiveService) CreateBaseTableArchive(ctx context.Context, req *adminv1.CreateBaseTableArchiveRequest) (*emptypb.Empty, error) {
	err := s.baseTableArchiveCase.CreateBaseTableArchive(ctx, req.GetBaseTableArchive())
	if err != nil {
		log.Error(fmt.Sprintf("CreateBaseTableArchive %v", err))
		return nil, errorsx.WrapInternal(err, "创建归档配置失败")
	}
	return new(emptypb.Empty), nil
}

// UpdateBaseTableArchive 更新表归档配置。
func (s *BaseTableArchiveService) UpdateBaseTableArchive(ctx context.Context, req *adminv1.UpdateBaseTableArchiveRequest) (*emptypb.Empty, error) {
	err := s.baseTableArchiveCase.UpdateBaseTableArchive(ctx, req.GetBaseTableArchive())
	if err != nil {
		log.Error(fmt.Sprintf("UpdateBaseTableArchive %v", err))
		return nil, errorsx.WrapInternal(err, "更新归档配置失败")
	}
	return new(emptypb.Empty), nil
}

// DeleteBaseTableArchive 删除表归档配置。
func (s *BaseTableArchiveService) DeleteBaseTableArchive(ctx context.Context, req *adminv1.DeleteBaseTableArchiveRequest) (*emptypb.Empty, error) {
	err := s.baseTableArchiveCase.DeleteBaseTableArchive(ctx, req.GetId())
	if err != nil {
		log.Error(fmt.Sprintf("DeleteBaseTableArchive %v", err))
		return nil, errorsx.WrapInternal(err, "删除归档配置失败")
	}
	return new(emptypb.Empty), nil
}

// SetBaseTableArchiveStatus 设置表归档配置状态。
func (s *BaseTableArchiveService) SetBaseTableArchiveStatus(ctx context.Context, req *adminv1.SetBaseTableArchiveStatusRequest) (*emptypb.Empty, error) {
	err := s.baseTableArchiveCase.SetBaseTableArchiveStatus(ctx, req)
	if err != nil {
		log.Error(fmt.Sprintf("SetBaseTableArchiveStatus %v", err))
		return nil, errorsx.WrapInternal(err, "设置归档配置状态失败")
	}
	return new(emptypb.Empty), nil
}
